.github/workflows/gradle.yml

```diff
@@ -237,14 +237,15 @@ jobs:
       - name: Ensure no file change
         run: git --no-pager diff && test -z "$(git --no-pager diff)"
 
-      - name: Generate Test Report
-        uses: dorny/test-reporter@v1
+      - name: Publish Test Results
+        uses: cpdeethree/publish-unit-test-result-action/composite@v0.0.13
```

Comment:
Correct. Essentially we are stripping the JSON that is generated from that job and POSTing it to check_runs API into the 'text' field so that it can be retrieved via Github API. That text field has a limit of 65k characters so there had to be a significant reduction of the output generated by the action.

This could normally be done outside of the context of the action itself because the job also writes a json file to the github action runner. Originally I intended to do this with a simple bash or python script that would read that json, strip it down, and then PATCH the checks run api with the change - this would have been IMO a somewhat cleaner solution. But Github wants you to set up and publish an App to perform this kind of work - it won't let you do it with just a PAT. Introducing an app just to PATCH one API felt like it was unnecessary and dirtier than doing it this way.

You can see the (minimal) changes here: https://github.com/EnricoMi/publish-unit-test-result-action/compare/master...cpdeethree:publish-unit-test-result-action:master

I'm not sure if it makes sense to mainline this eventually but it might be worth asking the maintainer if it makes sense to do so.

Reply:
After chatting with Davin further, we decided to instead going to attempt use the non-fork, bring in existing github app "octavia bot" to perform the PATCH, and lift the changes out of the forked action into a python script that lives in /tools/bin that reads/trims the existing JSON

Reply:
Unfortunately when I looked into this further it looks like the action in its current form does not allow you to expose individual test case results in the output json - it only has information for failed stats. I [filed an issue](https://github.com/EnricoMi/publish-unit-test-result-action/issues/365)to extend this, but in the meantime I'm going to revert to using my fork

Reply:
@davinchia well, it looks like the maintainer of that repo is super responsive and he helped me mainline my pull request to expose that data very quickly. So now I will switch this pr to do what I originally thought was the best solution that we discussed, which is to not use the fork and instead read the output of the action and PATCH the checks run api.

Reply:
amazing! thank you for being responsive to me :)